### ISSUE TITLE
The form of data augmentation in training

### ISSUE THREAD

**Issue by Zengyf-CVer** [ultralytics/yolov5#5613]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

@glenn-jocher 
In yolov5 training, is the data augmentation changed once per epoch or once per batch.

**Comment by glenn-jocher** [comment]
@Zengyf-CVer 👋 Hello! Thanks for asking about **image augmentation**. Augmentations are actually applied randomly per image/mosaic, i.e. so 16 random times in a batch if you use --batch-size 16.

YOLOv5 🚀 applies online imagespace and colorspace augmentations in the trainloader (but not the testloader) to present a new and unique augmented [Mosaic](https://github.com/ultralytics/yolov5/blob/90b7895d652c3bd3d361b2d6e9aee900fd67f5f7/utils/datasets.py#L678-L732) (original image + 3 random images) each time an image is loaded for training. **Images are never presented twice in the same way**.

![YOLOv5 augmentation](https://user-images.githubusercontent.com/26833433/120995721-f3cfed00-c785-11eb-8ee2-b6ef2fa205e8.jpg)

The hyperparameters used to define these augmentations are in your hyperparameter file (default `data/hyp.scratch.yaml`) defined when training:
```
python train.py --hyp hyp.scratch.yaml
```
https://github.com/ultralytics/yolov5/blob/90b7895d652c3bd3d361b2d6e9aee900fd67f5f7/data/hyp.scratch.yaml#L1-L33

You can view the effect of your augmentation policy in your train_batch*.jpg images once training starts. These images will be in your train logging directory, typically `yolov5/runs/train/exp`:

`train_batch0.jpg` shows train batch 0 mosaics and labels:  
> <img src="https://user-images.githubusercontent.com/26833433/131255960-b536647f-7c61-4f60-bbc5-cb2544d71b2a.jpg" width="700">

Good luck and let us know if you have any other questions!

**Comment by Zengyf-CVer** [comment]
@glenn-jocher 
Thank you very much.

**Comment by mauriceackel** [comment]
@glenn-jocher This means that if one wants to disable augmentation, they would simple set all the augmentation-related properties to `0.0`?

**Comment by glenn-jocher** [comment]
@mauriceackel yes that's correct!
